File: retroauth-android/src/test/java/com/andretietz/retroauth/AndroidMethodCacheTest.java
package com.andretietz.retroauth;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 23)
public class AndroidMethodCacheTest {

    private AndroidMethodCache androidMethodCache;

    @Before
    public void setup() {
        androidMethodCache = new AndroidMethodCache();
    }

    @Test
    public void shouldRegisterTokenTypeAndBeGettable() throws Exception {
        AndroidTokenType androidTokenType = new AndroidTokenType("accountType", "token1type");
        AndroidTokenType androidTokenType2 = new AndroidTokenType("accountType", "token2type");
        androidMethodCache.register(1, androidTokenType);
        androidMethodCache.register(2, androidTokenType2);

        assertNotNull(androidMethodCache.getTokenType(1));
        assertEquals(androidTokenType, androidMethodCache.getTokenType(1));
        assertNotNull(androidMethodCache.getTokenType(2));
        assertEquals(androidTokenType2, androidMethodCache.getTokenType(2));
    }
}

File: retroauth-android/src/test/java/com/andretietz/retroauth/AndroidMethodCacheTest.kt
//package com.andretietz.retroauth
//
//import org.junit.Assert.assertEquals
//import org.junit.Assert.assertNotNull
//import org.junit.Test
//import org.junit.runner.RunWith
//import org.robolectric.RobolectricTestRunner
//
//@RunWith(RobolectricTestRunner::class)
//class AndroidMethodCacheTest {
//
//    private val androidMethodCache: AndroidMethodCache = AndroidMethodCache()
//
//    @Test
//    @Throws(Exception::class)
//    fun shouldRegisterTokenTypeAndBeGettable() {
//        val androidTokenType = AndroidTokenType("accountType", "token1type")
//        val androidTokenType2 = AndroidTokenType("accountType", "token2type")
//        androidMethodCache.register(1, androidTokenType)
//        androidMethodCache.register(2, androidTokenType2)
//
//        assertNotNull(androidMethodCache.getTokenType(1))
//        assertEquals(androidTokenType, androidMethodCache.getTokenType(1))
//        assertNotNull(androidMethodCache.getTokenType(2))
//        assertEquals(androidTokenType2, androidMethodCache.getTokenType(2))
//    }
//}
